### PR TITLE
Address ON_WSL() bug #4310

### PR DESCRIPTION
**Pull request by JackofSpades707**
Solution for #4310
```python
return "microsoft" in platform.release().lower()
return "wsl" in platform.release().lower()
```
one of these may be a better solution depending on if the result of `platform.release()` varies by distro/wsl version

Output for ArchLinux WSL2:
`5.4.72-microsoft-standard-WSL2`

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

### CODE REVIEW

**Comment by JackofSpades707** [comment]
It also may make more sense to only display the 'on wsl' field if Linux is detected

**Comment by gforsyth** [comment]
Thanks for putting this in @JackofSpades707 !  Could you add a news entry describing the changes?  

https://xon.sh/devguide.html#changelog

File: xonsh/xonfig.py [hunk 1]
```diff
@@ -524,6 +524,7 @@ def _info(ns):
     )
     if ON_LINUX:
         data.append(("distro", linux_distro()))
+        data.append(("on wsl", bool(ON_WSL))),
```
Comment (gforsyth) [inline xonsh/xonfig.py:527]: This will need to be imported from `xonsh.platform`

Comment (JackofSpades707) [reply xonsh/xonfig.py:528]: Whoops, missed that, my bad.

**Comment by JackofSpades707** [comment]
@gforsyth 
Does that look better?

File: news/fix_wsl_detection.rst [hunk 2]
```diff
@@ -0,0 +1,13 @@
+**Added:**
+* 'on wsl' field when running xonfig (when linux is detected)
```
Comment (gforsyth) [inline news/fix_wsl_detection.rst:2]:
```suggestion
**Added:**

* ``on wsl`` field when running xonfig (when linux is detected)
```

RST failures bork the CI

File: news/fix_wsl_detection.rst [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+**Changed:**
+
+**Deprecated:**
+
+**Removed:**
+
+**Fixed:**
+* wsl detection works on archlinux wsl2 now (and hopefully everywhere)
```
Comment (gforsyth) [inline news/fix_wsl_detection.rst:11]:
```suggestion
**Fixed:**

* wsl detection works on archlinux wsl2 now (and hopefully everywhere)
```

**Review by gforsyth — CHANGES_REQUESTED** [review]

**Comment by gforsyth** [comment]
Thanks @JackofSpades707 !  And welcome to xonsh!

**Comment by JackofSpades707** [comment]
Thanks :)

**Pull request merged**